tests/test_entry_points.py

```diff
@@ -34,3 +79,54 @@ def test_entry_points_exist(self):
             dump.close()
             # check if the system return code is 0 (success)
             self.assertEqual(ex.exception.code, 0)
+
+    def test_match_template(self):
+        defaults = {
+            # 'pytom_match_template.py': '',
+            '-t': str(TEMPLATE),
+            '-m': str(MASK),
+            '-v': str(TOMOGRAM),
+            '-d': str(DESTINATION),
+            '--angular-search': '35',
+            '--tilt-angles': str(TILT_ANGLES),
+            '--per-tilt-weighting': '',
+            '--dose-accumulation': str(DOSE),
+            '--defocus': str(DEFOCUS_IMOD),
+            '--amplitude-contrast': '0.08',
+            '--spherical-aberration': '2.7',
+            '--voltage': '300',
+            '--tomogram-ctf-model': 'phase-flip',
+            '-g': '0',
+        }
+
+        def start(arg_dict):  # simplify run
+            entry_points.match_template(prep_argv(arg_dict))
+
+        # test valid defocus arguments
+        for z in [str(DEFOCUS_IMOD), str(DEFOCUS), '3000']:
+            arguments = defaults.copy()
+            arguments['--defocus'] = z
+            start(arguments)
+
+        # test faulty args
+        for z in ['asdf.txt', 'asdf']:
+            dump = StringIO()
+            with self.assertRaises(SystemExit) as ex, redirect_stdout(dump):
```

Comment:
you also have to dump stderr (with the import above
```suggestion
            with self.assertRaises(SystemExit) as ex, redirect_stdout(dump), redirect_stderr(dump):
```